src/control/lib/control/pool.go

```diff
@@ -1026,6 +1007,14 @@ func GetMaxPoolSize(ctx context.Context, rpcClient UnaryInvoker) (uint64, uint64
 		nvmeRanksBytes := make(map[system.Rank]uint64, 0)
 		for _, nvmeController := range hostStorage.NvmeDevices {
 			for _, smdDevice := range nvmeController.SmdDevices {
+				if !smdDevice.NvmeState.IsNormal() {
```

Comment:
I'm not sure if this is the correct course of action. In production, the admin will most likely never see debug logs. Additionally, I'm not sure that it's correct to be doing this kind of filtering on the client side. If a SMD device is in a bad state, it seems like it should either be filtered out on the server side, or else maybe the server-side RPC handler returns an error?

I don't know enough about the possibilities to have a strong opinion about exactly what to do, but I do think we probably shouldn't be silently proceeding if the storage state isn't normal...

Reply:
Currently it is not possible to filter the rank to use [DAOS-9557](https://daosio.atlassian.net/browse/DAOS-9557).
Until this last feature has been developed, I do not see better solution than printing informative message to the end user: with real system the probability of defunct devices is really high.
- [ ] Replace Debug message at line 1013 with Warning message.

Reply:
Agree this shouldn't have to be done on client side but state based filtering is not currently done on SMD devices belonging to a controller. Filtering with a state mask has been applied in `storage query list-devices`.
Thinking about how a user should be alerted if an SSD has faults, currently people are probably proactively running `storage query list-devices --show-evicted` but there really should be a RAS event raised when this happens. I think the warning is good for the moment but I wonder if we should also be reporting it on normal scan.

Reply:
NEW state can occur if there are more SSDs than necessary for the number of targets, this is kind of an edge case but isn't necessarily reporting a bad state.